### ISSUE TITLE
Cannot connect to influxdb on non default port

### ISSUE THREAD

**Issue by TomerAmirV** [influxdata/kapacitor#925]
We are using influxdb and kapacitor 0.13.0

We changed the default `[http]` port int the influxdb configuration to `8118` from `8086`.
I installed kapacitor, and changed the `urls` from:
`urls = ["http://localhost:8086"]`
to:
`urls = ["http://localhost:8118"]`

When I do that, the kapacitor log does not report on any error, and it also seems like it manages to subscribe to influxdb, but when I add ticks, I do not get the reports.

I tried to revert the ports back, and everything works.

Am I doing something wrong?
Did I miss another configuration that needs to be changed to the new port?

Thanks!

**Comment by nathanielc** [comment]
@TomerAmirV This should work without issue. The config you show is correct. Version 0.13 means you are using UDP ports, can you confirm that the UDP port is open on the Kapacitor host?

What is the output of `show subscriptions`?

**Comment by TomerAmirV** [comment]
Thanks!
The Kapacitor and Influx are installed on the same machine, but I validated anyway, and the UDP port is open (Managed to send udp packets)

When I enter `show subscriptions`, I see that kapacitor is subscribed to all of out databases both on `udp` and `http`.

Here is an example from one of the databases

```
default	"kapacitor-5a25f439-ab72-4ea1-b7f7-11d19cb6db12"	"ANY"	http://localhost:9092
default	"kapacitor-a44531ff-df1b-40a6-a3df-77f44ee02330"	"ANY"	udp://localhost:42821
```

**Comment by TomerAmirV** [comment]
Ok, Apparently the problem was something else.
When I removed all the subscriptions, and the restarted Kapacitor, it started working.
When does Kapacitor remove it's subscription?
Does influxDB support multiple subscriptions?

Could it be that since influxdb failed to send a report to  an older subscription, it did not send one to the new subscription?

How can I prevent this bug from happening again?

**Comment by nathanielc** [comment]
> How can I prevent this bug from happening again?

There was an `http` based subscription and you are using 0.13 which doesn't have support for `http` subcriptions. This probably means that you spun up 1.0 Kapacitor at one point and it created the `http` subscriptions. Kapacitor will clean up old subscriptions that have the same ID, but since you had two unique Kapacitor clusters IDs, the 0.13 Kapacitor didn't clean up the old http subscription since it was from a different Kapacitor instance.

In short, keep the versions in sync and make sure to preserve the cluster ID during an upgrade, or just clean up manually after an upgrade if you do change the cluster ID.
